### PR TITLE
Update meta.yaml

### PR DESCRIPTION
**Pull request by enriqueartal**
Added sirocco as a run dependency

Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

As suggested, I added sirocco as a run dependency, hope it works

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -161,6 +161,7 @@ requirements:
     - readline
     - {{ pin_compatible("rw", max_pin="x.x") }}
     - singular
+    - sirocco
```
Comment (saraedum) [inline recipe/meta.yaml:164]: It should be a `host` dependency, so `sagelib` is built against it. Not a `run dependency`.

**Comment by saraedum** [comment]
image
Did you bump the build number?

**Comment by saraedum** [comment]
Could you add a test that checks that sirocco actually was included? Maybe run the doctests of the sirocco.pyx file?

**Comment by saraedum** [comment]
Once that works, we could consider adding an [`ignore_run_exports`](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#export-runtime-requirements) to not bundle `sirocco` and instead have a `run_constrained` that sets sirocco with a `pin_compatible`. Not sure if that's the right approach though @isuruf.

File: recipe/meta.yaml (hunk 1 above)
Comment (enriqueartal) [reply recipe/meta.yaml:164]: OK, I'll change it

**Comment by enriqueartal** [comment]
> image Did you bump the build number?
Actually, not

**Comment by enriqueartal** [comment]
> 



>

**Comment by enriqueartal** [comment]
> Could you add a test that checks that sirocco actually was included? Maybe run the doctests of the sirocco.pyx file?

Actually, I do not know how to add this test

**Comment by enriqueartal** [comment]
> Once that works, we could consider adding an [`ignore_run_exports`](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#export-runtime-requirements) to not bundle `sirocco` and instead have a `run_constrained` that sets sirocco with a `pin_compatible`. Not sure if that's the right approach though @isuruf.

I closed by mistake and change from run to host, I do not know how to proceed with the other points.

**Comment by enriqueartal** [comment]
I changed sirocco from run to host and now the tests do not pass. Any ideas?

**Comment by saraedum** [comment]
The error message is

> nothing provides requested sirocco

Are the dependencies of sirocco incompatible with some of the other dependencies listed here?

**Comment by saraedum** [comment]
> > Could you add a test that checks that sirocco actually was included? Maybe run the doctests of the sirocco.pyx file?
> 
> Actually, I do not know how to add this test

There's this section in the reicpe:

```
  source_files:
    - src/sage/version.py
  imports:
    - sage
  commands:
    - sage -v
    - sage -c 1+1
    # too many doctests fail still but we test that the doctesting framework works by testing a trivial file
    - sage -tp src/sage/version.py
```

You should be able to add another `sage -tp` there. Make sure to provide the `sirocco.pyx` file in `source_files` as well.

**Comment by enriqueartal** [comment]
There is the requirement section for sirocco:
  ```
build:
        - {{ compiler('c') }}
        - {{ compiler('cxx') }}
    - autoconf
    - automake
    - libtool
    - m4
    - make
  host:
    - gmp
    - mpfr
  run:
```
Possibly some build options are not essential, I could try. But the complain is about `sirocco` not found. New try

**Comment by enriqueartal** [comment]
It complains again about sirocco not found, but it is in conda-forge, maybe some database is not updated?

**Comment by enriqueartal** [comment]
I am trying to test first in my computer; I have conda installed and added forge channel, in particular sirocco is found. I clone my fork of sagelib; is there a short howto in order to test locally this fork?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -181,13 +183,15 @@ test:
     - pip
   source_files:
     - src/sage/version.py
+    - src/sabe/libs/sirocco.pyx
```
Comment (isuruf) [inline recipe/meta.yaml:186]:
```suggestion
    - src/sage/libs/sirocco.pyx
```

**Comment by enriqueartal** [comment]
This seems to be the test error: `ModuleNotFoundError: No module named 'pkgconfig'`

File: recipe/build.sh [hunk 3]
```diff
@@ -27,7 +27,11 @@ rm -f build/pkgs/boost/spkg-configure.m4
 
 rm $PREFIX/bin/$HOST-pkg-config
 make configure
-./configure --prefix="$PREFIX" --with-python="$PYTHON"
+./configure \
+  --prefix="$PREFIX" \
+  --with-python="$PYTHON" \
+  --with-system-pkgconf=no \
```
Comment (isuruf) [inline recipe/build.sh:33]: Please remove this

Comment (enriqueartal) [reply recipe/build.sh]: Done. There are two type of errors, pkgconfig module not found and in some architectures, sirocco not found

Comment (enriqueartal) [reply recipe/build.sh]: Do you have any ideas how to solve it? I wonder if something must be changed in sirocco to work with other architectures. 

I have no clue how to deal with the problem of the pkgconfig module in the test.

**Comment by isuruf** [comment]
In about 2-3 hours, you'll be able to do `mamba install sage sagemath-sirocco` to get the sage extension that uses sirocco.

**Comment by enriqueartal** [comment]
I tested in binder (I replaced in the `environment.yml`  the lines `- sage = 9.6` and `- sirocco` by `- sagemath-sirocco`. I will update properly my repos. Thanks to Fernando @isuruf and Julian @saraedum. There were only some harmless warnings concerning some gap packages. I guess that building a sagemath-optional would be to give but it would be nice to have a way to contruct images with more than one optional packages. My python skills are not very good but if you have an idea i can collaborate.

**Pull request closed without merging**